### ISSUE TITLE
[PRE REVIEW]: pylustrator: code generation for reproducible figures for publication

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1783]
**Submitting author:** @rgerum (<a href="http://orcid.org/0000-0001-5893-2650">Richard Gerum</a>)
**Repository:** <a href="https://github.com/rgerum/pylustrator" target ="_blank">https://github.com/rgerum/pylustrator</a>
**Version:** v0.12.0
**Editor:** @labarba
**Reviewers:** @story645, @tacaswell

**Author instructions**

Thanks for submitting your paper to JOSS @rgerum. **Currently, there isn't an JOSS editor assigned** to your paper.

@rgerum if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```


**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01783/joss.01783/10.21105.joss.01783.pdf)

**Comment by kyleniemeyer** [comment]
@labarba can you edit this submission focused on reproducibility?

**Comment by labarba** [comment]
@whedon assign @labarba as editor

**Comment by whedon** [comment]
OK, the editor is @labarba

**Comment by labarba** [comment]
@rgerum — This is a re-submission of the same software that was considered out-of-scope on June 21:
https://github.com/openjournals/joss-reviews/issues/1499#issuecomment-504573661

Can you explain here what is different in the current submission, if anything, to the previous one?

**Comment by rgerum** [comment]
I rewrote the paper completely to stress more that the software is a tool in its own right and not just a small extension of the matplotlib package. Pylustrator is not a tool that extends the possible plotting functions of matplotlib, but is a tool to generate python code that composes single plots into complex composite figures for publication.
As I observed that most scientists I know compose their figures after plotting in image manipulation software (e.g. GIMP, Photoshop, etc.) which is error prone (accidentally altering the data points) and not reproducible (for other and also for oneself if the evaluation needs to be adapted later on). Out of this need I developed this software that aims to provide a reproducible tool to achieve the figure composition with code.
As the software is already used in a lot of publications, I want to have a citable paper and a way to further promote it, as I think it is a valuable contribution for science.

As I see other papers in JOSS that are also focusing on visualizations, e.g.
https://joss.theoj.org/papers/10.21105/joss.01425
https://joss.theoj.org/papers/10.21105/joss.01628
I do not see why my submission could not fit into JOSS.

**Comment by labarba** [comment]
👋  @ulfaslak — You published a visualization tool in JOSS a few weeks ago. Could you help us out now by reviewing this new submission?
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

**Comment by labarba** [comment]
👋 @tacaswell — Would you be able to review this submission for JOSS? Seems right up your alley... 
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

**Comment by labarba** [comment]
👋 @aokad — Are you interested in reviewing this JOSS submission? You signed up with "visualization" as your area of interest…
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

**Comment by aokad** [comment]
Sorry @labarba, I am unable to review at this time.

**Comment by ulfaslak** [comment]
Sorry for the slow response @labarba. I also cannot make time to review right now.

**Comment by labarba** [comment]
👋  @aokad — You have listed visualization as your topic for reviewing... would you be willing to review this submission for JOSS?
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

**Comment by labarba** [comment]
👋  @fil — I'm looking for reviewers for this JOSS submission... could you have a look and see if you may want to contribute a review?
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

**Comment by labarba** [comment]
👋  @alexanderfurnas — I notice you listed visualization as one of your review topics... would you be interested in contributing a JOSS review for this submission?
**pylustrator: code generation for reproducible figures for publication**
https://github.com/rgerum/pylustrator

(BTW, I found that you entered your handle as `zfurnas`, your Twitter handle rather than GitHub, which is probably why you haven't been asked to review all this time!)

**Comment by Fil** [comment]
I'm sorry I don't have the bandwidth and have to decline. I've watched the
demo video and it seems like a fine and useful tool.

**Comment by labarba** [comment]
To potential reviewers that said you are unable to review: please unsubscribe to this thread to save your inbox! 
@aokad : I just noticed that I pinged you twice, after you noted your unavailability... Oops, disregard! 

@tacaswell, @alexanderfurnas — let us know if you might be able to contribute a review for this JOSS submission. Thanks!

**Comment by labarba** [comment]
@whedon assign @story645 as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @story645

**Comment by labarba** [comment]
👋  @rgerum — Apologies for the slow-moving review here. I had a hard time finding willing reviewers, and then dropped the ball, I'm afraid. We now have one reviewer, and am looking for a second. Stay tuned!

Thank you for agreeing to review, @story645! Have a look at the [JOSS Reviewer Guidelines](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html) and let me know if you have any questions. We'll need to add a second reviewer before starting the Review issue, but in the meantime you can look through the software and start your assessment.

**Comment by tacaswell** [comment]
Sorry for the slow response, my github notifications are a dumpster fire.

I am willing to review this.

**Comment by labarba** [comment]
@whedon add @tacaswell as reviewer

**Comment by whedon** [comment]
OK, @tacaswell is now a reviewer

**Comment by labarba** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/1989. Feel free to close this issue now!
